src/swarm-topography/Swarm/Game/Scenario/Topography/Structure/Recognition/Type.hs

```diff
@@ -240,11 +236,24 @@ makeLenses ''RecognizerAutomatons
 --
 -- The two type parameters, `b` and `a`, correspond
 -- to 'Cell' and 'Entity', respectively.
-data FoundStructure b a = FoundStructure
-  { structureWithGrid :: StructureWithGrid b a
-  , upperLeftCorner :: Cosmic Location
+type FoundStructure b a = PositionedStructure (StructureWithGrid b a)
+
+-- | NOTE: A structure's name + orientation + position will uniquely
+-- identify it in the world.  Note that position alone is not sufficient;
+-- due to transparency, a completely intact smaller structure can co-exist
+-- within a larger structure, both sharing the same upper-left coordinate.
+-- However, two identical structures (with identical orientation) cannot
+-- occupy the same space.
```

Comment:
Is this a problem? 🤔 My understanding of structures is that they are compared structurally, so you can not tell if they have been placed at a location at the start or by the player.

Reply:
At the time that the game state is initialized, there is an alternate code path that just verifies the intactness of any structures that have been placed at a location at the start; a "search" is not needed since the location is already available.  After the game has started, however, there is no way to differentiate whether a structure had been pre-placed or placed by the player.